src/lib/pubkey/ecies/ecies.cpp

```diff
@@ -45,6 +45,11 @@ class ECIES_PrivateKey final : public EC_PrivateKey, public PK_Key_Agreement_Key
          return m_key.public_key();
          }
 
+      bool supports_operation(PublicKeyOperation op) const override
+         {
+         return (op == PublicKeyOperation::KeyAgreement);
+         }
```

Comment:
Oh now that I look at the code again I do think we want `KeyAgreement` here because `ECIES_PrivateKey` implements `create_key_agreement_op` but not the cooresponding encryption version.

`ECIES_PrivateKey` seems mostly to exist because our underlying ECDH always clears the cofactor and I guess there is some ECIES mode that does not. But this doesn't seem to matter since we don't really support elliptic curves with cofactors anyway?? Feel like all of this code could just be removed.